### PR TITLE
Jakarta Mail 2.0

### PR DESCRIPTION
**Pull request by lukasj**
## Specification PR template
When creating a specification project release review, create PRs with the content defined as follows.

Include the following in the PR:
- [x] A directory in the form wombat/x.y where x.y is the release major.minor version, and the directory contains the following.
- [x] Specification PDF in the form of wombat-spec-x.y.pdf
- [x] Specification HTML in the form of wombat-spec-x.y.html
- [x] A specification page named _index.md following the template at:
      https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md
- [x] For a Progress Review, that sufficient progress has been made on a Compatible Implementation and TCK, to ensure that the spec is implementable and testable.
- [x] For a Release Review, a summary that a Compatible Implementation is complete, passes the TCK, and that the TCK includes sufficient coverage of the specification.
The TCK users guide MUST include the instructions to run the compatible implementations used to validate the release.
Instructions MAY be by reference.
- [x] The URL of the OSSRH staging repository for the api, javadoc:
      https://jakarta.oss.sonatype.org/content/groups/staging/jakarta/mail/jakarta.mail-api/2.0.0/
- [x] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:
      https://download.eclipse.org/ee4j/mail/tck/eftl/jakarta-mail-tck-2.0.0.zip
- [x] The URL of the compatibility certification request issue:
      https://github.com/eclipse-ee4j/mail/issues/470
- [x] Specification JavaDoc in the wombat/x.y/apidocs directory. 
If desired, an optional second PR can be created to contain just the JavaDoc in the `apidocs` directory.

### CODE REVIEW

**Comment by edbratt** [comment]
For the Specification Committee -- we are using the following detailed checklist:
# Spec Review Checklist

1. Spec PR
  - [ ] PR uses [template](https://github.com/jakartaee/specifications/blob/master/pull_request_template.md)
  - [ ] Directory of form {spec}/x.y
  - [ ] PDF of form {spec}-spec-x.y.pdf ("-spec" preferred but not required)
  - [ ] HTML of form {spec}-spec-x.y.html ("-spec" preferred but not required)
  - [ ] Index page {spec}/x.y/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md)
  - [ ] Index page {spec}/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_index_template.md)
  - [ ] No other files (e.g., no jakarta_ee_logo_schooner_color_stacked_default.png)
  - [ ] Staging repository link of the form https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/{spec}/jakarta.{spec}-api/x.y.z/
  - [ ] EFTL TCK link of the form http://download.eclipse.org/.../+.zip
  - [ ] Compatibility certification link of the form https://github.com/eclipse-ee4j/{project}/#{issue}
  - [ ] (Optional) Second PR for just apidocs

2. _index.md
  - [ ] Link to spec pdf
  - [ ] Link to spec html
  - [ ] Link to apidocs
  - [ ] Link to final TCK download zip file of the form https://download.eclipse.org/jakartaee/{spec}/x.y/*{spec}-tck-x.y.z.zip
  - [ ] Link to API jar file of the form https://search.maven.org/artifact/jakarta.{spec}/jakarta.{spec}-api/x.y.z/jar
  - [ ] Name of and link to at least one Compatible Implementation

3. javadocs
  - [ ] Footer contains Eclipse copyright and link to license
  - [ ] ESFL license is included, usually as doc-files/speclicense.html
  - [ ] no META-INF directory in PR
  - [ ] javadocs-jar artifact matches apidocs (optional for this release)

4. Spec PDF
  - [ ] Correct spec title
  - [ ] Version number of the form x.y, not x.y.z
  - [ ] Correct Eclipse copyright line
  - [ ] No DRAFT or SNAPSHOT
  - [ ] Correct Logo

5. Spec HTML
  - [ ] Same as PDF

6. TCK zip file
  - [ ] README file (optional for this release)
  - [ ] EFTL license file, preferably named LICENSE.md
  - [ ] User's Guide (or equivalent documentation)
  - [ ] How to test the Compatible Implementation(s) listed in _index.md above with the TCK (may be in UG)

7. TCK User's Guide (or equivalent documentation)
  - [ ] Software requirements listed
  - [ ] Installation and configuration described
  - [ ] How to run tests
  - [ ] Where to file challenges

8. Compatibility certification request
  - [ ] Request follows [template](https://github.com/jakartaee/specification-committee/blob/master/compatibility-certification-request.md)
  - [ ] SHA-256 fingerprint matches staged TCK zip file
  - [ ] Request issue has `certification` label.

9. TCK results summary
  - [ ] Page is hosted by Compatible Implementation project
  - [ ] Includes all information from certification request
  - [ ] Summary includes number of tests passed, failed, errors
  - [ ] SHA-256 fingerprint matches staged TCK zip file on cert request
 
10. Update Jakarta EE API jar
  - [ ] Update the Jakarta EE API jar by submitting a PR to the jakartaee-api project that updates the version number of your API jar file.

**Comment by smillidge** [comment]
I tried to incorporate the staged jar into GlassFish but it didn't work see https://github.com/eclipse-ee4j/mail/issues/465

**Comment by keilw** [comment]
@edbratt, @kwsutter, is the original checklist OK or should @lukasj use the one suggested below? I am reviewer/mentor here, so I assigned myself to review it. Would love to conduct it, but there are e.g. still checkboxes missing. And how does the issue @smillidge mentioned in Glassfish affect it?

**Comment by lukasj** [comment]
@keilw it is a reviewer how is supposed to use the checklist added by Ed

**Comment by keilw** [comment]
@lukasj But which of them, should I continue with the one in the body or do we have to replace it with a much more extensive one suggested below?

**Comment by lukasj** [comment]
@keilw the one titled `Spec Review Checklist` - that is the only checklist added by Ed I can see. What exactly do you mean by `the one below`?

**Comment by keilw** [comment]
@lukasj The "Specification PR template" is the one below, the topmost list was called "Specification PR template" so I'm wondering, if we should replace that? What was already checked seems valid or should we/I verify that again? I'm not sure, if I can personally stage a JAR like @smillidge tried, but I guess with a little help from others that is also possible.

**Comment by lukasj** [comment]
@keilw Why do you want to stage the jar yourself? Why do you think the current one (~3hours old) is wrong? The issue @smillidge had is supposed to be fixed.

**Comment by smillidge** [comment]
@lukasj thanks I tested the new staged jar and the error is fixed

**Comment by keilw** [comment]
@smillidge Thanks for confirming it. Btw. as Amelia also makes a big fuzz about "Eclipse or not Eclipse" in the names, I am confident for the compatible implementations "Eclipse Jakarta Mail 2.0.0" sounds fine, I did not see it related to the spec or API. This is one of the specs where API and implementation are not so clearly separated (also similar for JSONP) which IMO justifies the use of "Eclipse" in the project name, similar to most other compatible implementations.

**Comment by kwsutter** [comment]
@keilw The noise that is being made about "Eclipse Jakarta" vs "Jakarta" is just that - noise.  Do not allow that discussion to affect your reviewing/mentoring of these PRs.  Thanks!

Also, the Spec Checklist that Ed posted is the accurate one.  You as the reviewer/mentor should be checking off the items as you have confirmed that they are completed.  Lukas can use it as a reminder of what's to be expected.  But, the actual marking off of the checkboxes should be your responsibility.

**Comment by jeanouii** [comment]
As discussed yesterday during the platform call, I looked at the PR and looks like there are still javax mentions in code snipets.

```
grep javax mail-spec-2.0.html
<span class="keyword">import</span> <span class="include">javax.mail</span>.*;
<span class="keyword">import</span> <span class="include">javax.mail.event</span>.*;
<span class="keyword">import</span> <span class="include">javax.mail.internet</span>.*;
<span class="keyword">import</span> <span class="include">javax.mail</span>.*;
<span class="keyword">import</span> <span class="include">javax.mail</span>.*;
<span class="keyword">import</span> <span class="include">javax.mail.internet</span>.*;
<span class="keyword">import</span> <span class="include">javax.mail.search</span>.*;
<span class="keyword">import</span> <span class="include">javax.activation</span>.*;
<span class="keyword">import</span> <span class="include">javax.mail</span>.*;
<span class="keyword">import</span> <span class="include">javax.mail.event</span>.*;
<span class="keyword">import</span> <span class="include">javax.activation</span>.*;
<span class="keyword">import</span> <span class="include">javax.mail</span>.*;
<span class="keyword">import</span> <span class="include">javax.mail.internet</span>.*;
<span class="keyword">import</span> <span class="include">javax.activation</span>.*;
```

Nothing so far in apidocs.

File: mail/2.0/_index.md
Comment (jeanouii) [inline mail/2.0/_index.md:11]: shouldn't it have jakarta- in the file name?
Same for html bellow

**Review by jeanouii — COMMENTED** [review]
See comment in the file names

File: mail/2.0/_index.md
Comment (lukasj) [reply mail/2.0/_index.md]: PDF of form {spec}-spec-x.y.pdf ("-spec" preferred but not required) is the requirement, where spec is "mail" in this case. `jakarta-` prefix is not mandatory thing for this release, even though I've added it.

**Comment by lukasj** [comment]
feedback addressed

**Comment by lukasj** [comment]
per [the plan](https://eclipse-ee4j.github.io/jakartaee-platform/jakartaee9/JakartaEE9ReleasePlan) this should be in wave2 and not in wave0  (this **depends on**  activation which is in wave1). Someone wants to fix that?

**Comment by keilw** [comment]
@lukasj @jeanouii Great thanks, the PR has been updated accordingly I assume. I don't even think there's a bullet point for that in one of the lists.

**Comment by lukasj** [comment]
@keilw when do you think you're going to be done with the review of this and mark this as ready for ballot? I need to know that date in order to fix the release date for this spec in the _index.md_; current date there is 2020-08-25 which is in the past already. New date is going to be the date you provide + 3-4 weeks (ballot takes 2 weeks, release review takes a week, one needs time to submit stuff in between etc). Thanks.

**Comment by ivargrimstad** [comment]
> @keilw when do you think you're going to be done with the review of this and mark this as ready for ballot? I need to know that date in order to fix the release date for this spec in the _index.md_; current date there is 2020-08-25 which is in the past already. New date is going to be the date you provide + 3-4 weeks (ballot takes 2 weeks, release review takes a week, one needs time to submit stuff in between etc). Thanks.

@lukasj The date field in the _index.md_ is the Hugo publish date. It is ok if it is in the past since it is only used to determine if the page is showed or not, never displayed in any way. I just talked with @keilw and he hopes to get the review done today. I have looked over it, and everything seems good to go.

**Comment by lukasj** [comment]
@ivargrimstad thanks, good to know to not be worry about index. Is there any automation to handle changes on project pages, like https://projects.eclipse.org/projects/ee4j.mail/releases/2.0.0, too? That one currently says June 30, 2020

**Comment by ivargrimstad** [comment]
> @ivargrimstad thanks, good to know to not be worried about index. Is there any automation to handle changes on project pages, like https://projects.eclipse.org/projects/ee4j.mail/releases/2.0.0, too? That one currently says June 30, 2020

@lukasj That one is updated manually by logging in and editing the release record. I think that is the only way...

**Comment by keilw** [comment]
@lukasj @ivargrimstad The link to the TCK [https://download.eclipse.org/jakartaee/mail/2.0/jakarta-mail-tck-2.0.0.zip](https://download.eclipse.org/jakartaee/mail/2.0/jakarta-mail-tck-2.0.0.zip) does not seem to work.

**Comment by keilw** [comment]
@lukasj @ivargrimstad Not entirely sure about the copyright, it says
"Copyright (c) 2020 Eclipse Foundation." Mail 1.6 had the same with 2019, is it legitimate to drop the 2019 or should it say
"Copyright (c) 2019-2020 Eclipse Foundation." similar to what Bean Validation does?

**Comment by lukasj** [comment]
@keilw for links see [kevin's answer](https://github.com/jakartaee/specifications/pull/231#issuecomment-680253888), for cp year - "2019-2020" is definitely wrong, unless [the handbook](https://www.eclipse.org/projects/handbook/#ip-copyright-headers) does not contain up-to-date info wrt:

> ...The date may optionally be set to a range of years with the first and last years of the range *separated by a comma* (e.g. “2004, 2017”); ...

I fixed it to "2019, 2020" here.

**Comment by keilw** [comment]
@lukasj @ivargrimstad Looks good now, what about 10, who does that, I can't see a PR in https://github.com/eclipse-ee4j/mail/pulls that matches the description.

**Comment by lukasj** [comment]
@keilw see [mail from kevin](https://www.eclipse.org/lists/jakartaee-platform-dev/msg02154.html) or search through [all PRs](https://github.com/eclipse-ee4j/jakartaee-api/pull/75), not only open ones. Some people are pretty fast...

**Comment by keilw** [comment]
@lukasj Ok that was merged, thanks, I will merge this one and trigger the ballot once I am back from a dinner/concert. Also have to see what Kevin wrote about merging these PRS.

**Comment by lukasj** [comment]
@keilw I believe the process is to approve -> start ballot -> wait for result -> merge/dismiss

**Comment by lukasj** [comment]
@keilw note that this depends on activation (#172). It's good to see we're both done here and only waiting;-)

**Comment by ivargrimstad** [comment]
> @keilw I believe the process is to approve -> start ballot -> wait for result -> merge/dismiss

Yes, @lukasj is right! **Do not merge until the ballot has completed.**
@keilw: The next step is to initiate the ballot using the [initiate ballot email template](https://t.co/Ezbsdeuz33).

**Comment by keilw** [comment]
Great thanks, that'll make it easy for now.

**Review by keilw — APPROVED** [review]

**Comment by keilw** [comment]
Ballot complete!

Few more steps now. Spec committee mentor (myself) need to:
- [ ] adds this final checklist to the main PR.
- [ ] adds the approved label to the PRs, and sends out the Ballot Summary per this template to the public Jakarta EE Specification Committee email list
- [ ] calculates the staged EFTL TCK signature and promotes it to the committee download area using the https://ci.eclipse.org/jakartaee-spec-committee/job/promote-release/ job.
- [ ] updates the specification page with the ballot results.
- [ ] merges the specification (and apidocs) PRs, ensuring the "date:" field in the _index.md file has an appropriate value to allow publishing.
- [ ] notifies the EMO of the ballot results by email to emo@eclipse-foundation.org
- [ ] creates an issue in the specification project that includes the following checklist for the specification project team:
  - [ ] promotes api staging release promotes the specification api jars to maven central. An example release job script can be found here https://wiki.eclipse.org/MavenReleaseScript.
  - [ ] go through the merged jakarta.ee specification website page to verify all the links are valid.
  - [ ] approve the compatibility request.
  - [ ] The compatible implementation project/vendor MUST send an email to tck@eclipse.org for approval of the compatible implementation for trademark usage.
  - [ ] merge any final release branch as appropriate for the branch management for the project.

**Pull request merged**
